chef-config/lib/chef-config/mixin/fuzzy_hostname_matcher.rb

```diff
@@ -21,7 +21,7 @@ module Mixin
     module FuzzyHostnameMatcher
 
       def fuzzy_hostname_match_any?(hostname, matches)
-        if (!hostname.nil?) && (!matches.nil?)
+        unless hostname.nil? || matches.nil?
```

Comment:
How about `if hostname && matches`?

Reply:
I know historically we've battled over using .nil? or not. I was trying to maintained the .nil? behavior while untangling the knot in my brain.

Reply:
if there's no ambiguity over false being valid and nil meaning "invalid value that deliberately needs to avoid triggering on false" then dropping the `nil?` check reads better.

the only reason to keep that is worry over false.

Reply:
Updates to plain hostname/matches. For our use nil means an env var wasn't there and that's unset so there's no match.